### PR TITLE
[backend/frontend] Add Case tasks and case template (#issue/2846)

### LINKED ISSUE #2846: Implement tasks and tasks template in cases
## Use case

* Create a new type of entity "Task"
  * Name / Description / Due Date / Workflow
* In the settings of the platform, be able to pre-defined list of tasks (aka tasks list template)
* ~~In the Case menu, add a top menu "Tasks" to list all tasks~~
* In a Case, display the list of the tasks with the ability to:
  * Apply a task list template
  * Create a new task
* When creating a case, be able to select a task list template at creation to be applied automatically

### PR DESCRIPTION
**Pull request by Kedae**

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLine.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    paddingRight: 5,
+  },
+  goIcon: {
+    position: 'absolute',
+    right: -10,
+  },
+  itemIconDisabled: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLine.tsx]: Not used ?

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  },
+  itemIconDisabled: {
+    color: theme.palette.grey?.[700],
+  },
+  placeholder: {
+    display: 'inline-block',
+    height: '1em',
+    backgroundColor: theme.palette.grey?.[700],
+  },
+  paper: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx]: itemIconDisabled, placeholder and paper classes are seems to be not used

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+              style={{ width: value.width }}
+            >
+              {value.render?.(task, { fldt })}
+            </div>
+          ))}
+        </div>
+      }
+      />
+      <ListItemIcon classes={{ root: classes.goIcon }}>
+        <CaseTasksPopover task={task} paginationOptions={paginationOptions} />
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx]: Why we have popover here instead of arrow (like the others screens) ?

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    float: 'left',
+  },
+  item: {
+    paddingLeft: 10,
+    height: 50,
+  },
+  itemIcon: {
+    color: theme.palette.primary.main,
+  },
+  bodyItem: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx]: goIcon, item, itemIcon, bodyItem and chipInList classes seems to be not used

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksPopover.tsx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    top: 12,
+    left: 5,
+    color: 'inherit',
+  },
+  formContainer: {
+    padding: '10px 20px 20px 20px',
+  },
+}));
+
+const Transition = React.forwardRef((props: SlideProps, ref) => (
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksPopover.tsx]: You can use the Transition component

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksPopover.tsx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+  const handleOpen = (event: React.SyntheticEvent) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleOpenDelete = () => {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksPopover.tsx]: There are a hook to handle it
Should be used it ? -> useDeletion

File: opencti-platform/opencti-front/src/private/components/cases/CaseUtils.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
+import { graphql, ConnectionHandler, Variables } from 'react-relay';
+
+export const generateConnectionId = ({ recordId, key, params }: { recordId?: string, key: string, params?: Variables }) => ConnectionHandler.getConnectionID(recordId ?? 'root', key, params ?? {});
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseUtils.tsx:3]: Should be in the store.js file ?

**Comment by Kedae** [comment]
Following demo:

1. Need to update the CaseTemplate Creation/Edition to upgrade the usage of TaskTemplate
2. "Apply template" instead of attribute in Case
3. Keep history of caseTemplate updates in cases
4. Task should inherit marking definition and be able to be modified

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+              className={classes.bodyItem}
+              style={{ width: value.width }}
+            >
+              {value.render?.(task, { fldt })}
+            </div>
+          ))}
+        </div>
+      }
+      />
+      <ListItemIcon classes={{ root: classes.goIcon }}>
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLine.tsx:161]: where is this class `goIcon` ? There also seem to be unused classes in this file : `itemIconDisabled`, `paper`, `placeholder`

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+                      key: 'Pagination_caseTasks',
+                      params: tasksFilters,
+                    })],
+                  },
+                  onCompleted: () => {
+                    setSubmitting(false);
+                    setOpenCaseTemplate(false);
+                  },
+                  onError: (error) => {
+                    MESSAGING$.notifyError((error as unknown as RelayError).res.errors.at(0)?.data.reason);
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx]: why not use handleErrorInForm ?

**Comment by Archidoit** [comment]
- In Case Incident > Overview: Assignee is not clickable
image
- not clear error message when editing a task with no due date
image
- from Case Incident > Overview: create a task without changing the default due date. When clicking 'create', the task is created but with no due date.
image



@Kedae:
- Assignee not clickable in the platform
- DueDate is now nullable
- Default value for task is fixed

**Comment by Archidoit** [comment]
Proposal: Put column names in Case Overview>Tasks
image
like it is done in Group overview > members:
image


@Kedae @SarahBocognano 
--> Done

**Comment by Archidoit** [comment]
- In case template, it is possible to have several tasks with the same name:
image
- strange icon on the right after adding a task in a case template:
image
- Create 2 templates with the same task. Go to one of the templates and delete the task. Go to the other template: the task is not deleted unless you refresh the page. If you don't refresh and unlink the task, the popover appears, click 'Unlink': Nothing happens and the popover is grey.
image
- it is not possible to edit case template after creation (ex: change its description). 
- we can change case templates properties via the tool bar (ex: replace the label of a case template), but these properties cannot be seen anywhere.
- in Label&Attributes > Case template: the 'name' column name should be aligned with the templates names, not the templates icons:
image


@Kedae @SarahBocognano 
- Same name is normal and wanted :)
- Icon fixed
- This is a standard behavior of relay, as we do not refetch everytime
- Edition of case template done
- Toolbar fixed
- Alignment fixed

**Comment by Kedae** [comment]
All comments have been answered

**Comment by Archidoit** [comment]
- In Labels&Attributes > Case template > task: column name 'description' is not aligned
image
- idem in Case template
image
- Click on a case template to see its task > click on the 'description' column (to sort by description) > error occurs
image
- Select case templates via the checkbox, click on the delete button of the tool bar, the case template are not deleted (but their tasks are deleted)
image

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+      <div className="clearfix" />
+      <Paper classes={{ root: classes.paper }} variant="outlined">
+        {data.caseTasks && data.caseTasks.edges.length > 0 ? (
+          <div>
+            <CaseTasksLineTitles />
+            <List>
+              {data.caseTasks.edges.map(({ node }) => {
+                return (
+                  <CaseTasksLine
+                    key={JSON.stringify(node)} data={node} paginationOptions={paginationOptions}
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx]: Why not add the id in the fragment and use it here?

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+                    caseId={caseId}
+                  />
+                );
+              })}
+            </List>
+          </div>
+        ) : (
+          <div
+            style={{
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx]: Use classes.[className] ?

File: opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx [hunk 12]
```diff
@@ -0,0 +1,334 @@
...
+              display: 'table',
+              height: '100%',
+              width: '100%',
+              paddingTop: 15,
+              paddingBottom: 15,
+            }}
+          >
+          <span
+            style={{
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/CaseTasksLines.tsx]: Use classes.[className] ?

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx [hunk 13]
```diff
@@ -213,7 +214,7 @@ CaseRfiEditionOverviewProps
   const handleSubmitField = (name: string, value: Option | string | string[] | number | number[] | null) => {
     if (!enableReferences) {
       let finalValue: unknown = value as string;
-      if (name === 'x_opencti_workflow_id') {
+      if (['x_opencti_workflow_id'].includes(name)) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx:217]: Why this changes ?
It's more efficient ?

File: opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftDetails.tsx [hunk 14]
```diff
@@ -245,40 +245,40 @@ const CaseRftDetails: FunctionComponent<CaseRftDetailsProps> = ({
...
+                        <div className={classes.itemText}>
+                          {relatedContainer?.name}
+                        </div>
                       }
-                      limit={1}
                     />
-                  </div>
-                </ListItem>
+                    <div className={classes.itemAuthor}>
+                      {R.pathOr('', ['createdBy', 'name'], relatedContainer)}
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftDetails.tsx]: Ramda (if you have time and faith

File: opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+          <IconButton
+            aria-label="Close"
+            className={classes.closeButton}
+            onClick={() => setOpenUpdate(false)}
+            size="large"
+            color="primary"
+          >
+            <Close fontSize="small" color="primary" />
+          </IconButton>
+          <Typography variant="h6" classes={{ root: classes.title }}>
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx]: Class title not declared in classes

File: opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+    top: 12,
+    left: 5,
+    color: 'inherit',
+  },
+  formContainer: {
+    padding: '10px 20px 20px 20px',
+  },
+}));
+
+const Transition = React.forwardRef((props: SlideProps, ref) => (
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx]: Should be replaced by the Transition component

File: opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+  paginationOptions,
+  caseId,
+}) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const [commit] = useMutation(caseTasksPopoverDeletionMutation);
+
+  const [anchorEl, setAnchorEl] = useState<Element | null>(null);
+  const [displayDelete, setDisplayDelete] = useState(false);
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/case_task/CaseTasksPopover.tsx]: Could be replaced by the useDeletion hook ?

File: opencti-platform/opencti-front/src/private/components/cases/incidents/CaseIncidentEditionOverview.tsx [hunk 18]
```diff
@@ -140,15 +141,14 @@ const caseIncidentMutationRelationDelete = graphql`
...
-  context:
-  | readonly ({
-    readonly focusOn: string | null;
-    readonly name: string;
-  } | null)[]
-  | null;
-  enableReferences?: boolean;
-  handleClose: () => void;
+  caseRef: CaseIncidentEditionOverview_case$key
+  context: ReadonlyArray<{
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/cases/incidents/CaseIncidentEditionOverview.tsx:145]: Should we have a class to handle this type for all the edition component ? (in edition.js file)

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+    backgroundColor: theme.palette.background.nav,
+    padding: '20px 20px 20px 60px',
+  },
+  closeButton: {
+    position: 'absolute',
+    top: 12,
+    left: 5,
+    color: 'inherit',
+  },
+  importButton: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx]: Not used

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+  mutation CaseTemplateCreationMutation($input: CaseTemplateAddInput!) {
+    caseTemplateAdd(input: $input) {
+      ...CaseTemplateLine_node
+    }
+  }
+`;
+
+const caseTemplateValidation = (t: (name: string | object) => string) => Yup.object().shape({
+  name: Yup.string().required(t('This field is required')),
+  description: Yup.string(),
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx]: Could be nullable ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx [hunk 21]
```diff
@@ -0,0 +1,218 @@
...
+  tasks: Yup.array(),
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx:75]: Could be nullable ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateEdition.tsx [hunk 22]
```diff
@@ -0,0 +1,265 @@
...
+interface CaseTempateEditionProps {
+  existingTasks: Option[]
+  paginationOptions: CaseTemplateTasksLines_DataQuery$variables
+  queryRef: PreloadedQuery<CaseTemplateEditionQuery>
+}
+
+const caseTemplateValidation = (t: (name: string | object) => string) => Yup.object().shape({
+  name: Yup.string().required(t('This field is required')),
+  description: Yup.string(),
+  tasks: Yup.array(),
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateEdition.tsx:110]: Description & tasks could be nullable ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLine.tsx [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+    color: theme.palette.grey?.[700],
+  },
+  placeholder: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLine.tsx]: Not used ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLineDummy.tsx [hunk 24]
```diff
@@ -0,0 +1,86 @@
...
+    paddingRight: 5,
+  },
+  goIcon: {
+    position: 'absolute',
+    right: -10,
+  },
+  itemIconDisabled: {
+    color: theme.palette.grey?.[700],
+  },
+  placeholder: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateLineDummy.tsx]: Not used ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateTasksLine.tsx [hunk 25]
```diff
@@ -0,0 +1,153 @@
...
+    paddingRight: 5,
+  },
+  goIcon: {
+    position: 'absolute',
+    right: -10,
+  },
+  itemIconDisabled: {
+    color: theme.palette.grey?.[700],
+  },
+  placeholder: {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateTasksLine.tsx]: Not used ?

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateTasksPopover.tsx [hunk 26]
```diff
@@ -0,0 +1,248 @@
...
+  paginationOptions,
+  task,
+}) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const { caseTemplateId } = useParams() as { caseTemplateId: string };
+
+  const [anchorEl, setAnchorEl] = useState<PopoverProps['anchorEl']>(null);
+  const [displayUpdate, setDisplayUpdate] = useState<boolean>(false);
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateTasksPopover.tsx:90]: useDeletion hook ?

File: opencti-platform/opencti-front/src/relay/relayTypes.ts [hunk 27]
```diff
@@ -0,0 +1,10 @@
+export interface RelayError {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/relay/relayTypes.ts]: It's used ?

File: opencti-platform/opencti-front/src/utils/edition.js [hunk 28]
```diff
@@ -46,3 +46,7 @@ export const convertExternalReferences = (element) => (element?.externalReferenc
 export const convertCreatedBy = (element) => (isEmptyField(element?.createdBy)
   ? ''
   : { label: element.createdBy.name, value: element.createdBy.id });
+
+export const convertGenericOption = (element) => (isEmptyField(element)
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/utils/edition.js]: It's used ?

File: opencti-platform/opencti-front/src/utils/Entity.ts [hunk 29]
```diff
@@ -72,6 +72,8 @@ export const resolveLink = (type: string): string | null => {
       return '/dashboard/cases/rfis';
     case 'Case-Rft':
       return '/dashboard/cases/rfts';
+    case 'Case-Task':
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/utils/Entity.ts:75]: I don't understand why we need to add this line ?

File: opencti-platform/opencti-graphql/src/modules/case/case-converter.ts [hunk 30]
```diff
@@ -1,14 +1,13 @@
...
 import { buildStixDomain, cleanObject } from '../../database/stix-converter';
-import { STIX_EXT_OCTI } from '../../types/stix-extensions';
 import { INPUT_OBJECTS } from '../../schema/general';
+import { STIX_EXT_OCTI } from '../../types/stix-extensions';
+import type { StixCase, StoreEntityCase } from './case-types';
 
 const convertCaseToStix = (instance: StoreEntityCase): StixCase => {
   const cases = buildStixDomain(instance);
   return {
+    ...instance,
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/case/case-converter.ts:9]: For me, it's weird to add this line. This is not the same type at all.

File: opencti-platform/opencti-graphql/src/modules/case/case-task/case-task.ts [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+import caseTaskResolvers from './case-task-resolvers';
+import type { StixCaseTask, StoreEntityCaseTask } from './case-task-types';
+import { ENTITY_TYPE_CONTAINER_CASE_TASK } from './case-task-types';
+import caseTaskTypeDefs from './case-task.graphql';
+
+const CASE_TASK_DEFINITION: ModuleDefinition<StoreEntityCaseTask, StixCaseTask> = {
+  type: {
+    id: 'case-task',
+    name: ENTITY_TYPE_CONTAINER_CASE_TASK,
+    category: ENTITY_TYPE_CONTAINER
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/modules/case/case-task/case-task.ts:16]: Does it appear in the list of entity settings?
Is this what we want? (we can customize workflow)

File: opencti-platform/opencti-front/src/private/components/cases/CaseUtils.tsx (hunk 7 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/cases/CaseUtils.tsx:3]: No, because I don't think this should be extended right now as it a particular case. If in the future we have a similar behavior we will move it and talk about it in dev team :)

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx (hunk 13 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx:217]: No :) I changed it because I had a second item in the list at a moment, but not anymore. It doesn't change anything so I choose not to revert.

File: opencti-platform/opencti-front/src/private/components/cases/incidents/CaseIncidentEditionOverview.tsx (hunk 18 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/cases/incidents/CaseIncidentEditionOverview.tsx:145]: Edition is js now, not TS so we can't. In the future we should have that yes.

File: opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx (hunk 21 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/private/components/settings/case_templates/CaseTemplateCreation.tsx:75]: No, we choose to have an array of empty elements

File: opencti-platform/opencti-front/src/utils/Entity.ts (hunk 29 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/utils/Entity.ts:75]: Because for now we don't have a screen to see a task, but we list them in entities. We should to disable click

File: opencti-platform/opencti-graphql/src/modules/case/case-converter.ts (hunk 30 above)
Comment (Kedae) [reply opencti-platform/opencti-graphql/src/modules/case/case-converter.ts:9]: Not sure to get your point ? We are getting its attributes, it is the same as :
{
  name: instance.name,
  description: instance.description
}

What bothers you ?

File: opencti-platform/opencti-graphql/src/modules/case/case-task/case-task.ts (hunk 31 above)
Comment (Kedae) [reply opencti-platform/opencti-graphql/src/modules/case/case-task/case-task.ts:16]: Yes ? We want to customize workflow of tasks

**Comment by Kedae** [comment]
All comments except one alignment issue has been taken care of. Still investigating the alignment but issue can be reviewed/merged without it.

File: opencti-platform/opencti-graphql/src/modules/case/case-converter.ts (hunk 30 above)
Comment (RomuDeuxfois) [reply opencti-platform/opencti-graphql/src/modules/case/case-converter.ts:9]: Hum, true
I was just confused on first reading

**Comment by RomuDeuxfois** [comment]
I18n :
image
image

Style :
Miss pointer on line
image

QS :
what is the difference between unlink and delete? (In the settings)

----------


@SarahBocognano : 
Thank you for your feedback.
The diference between unlink and delete is that you can have the same task in multiple case template. So you can just unlink it but not delete it completly from the platform.

**Review by RomuDeuxfois — APPROVED** [review]

**Comment by Kedae** [comment]
After investigating, we need to have a simple detail page for Task to have Entities/Notifications/Dashboard to work correctly
